gems/quilt_rails/test/controllers/routing_test.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require "test_helper"
+require "action_controller"
+
+module Quilt
+  class RoutingTest < ActionDispatch::IntegrationTest
+    include ActiveSupport::Testing::Isolation
+
+    class PerformanceReportController < ActionController::Base; end
+    class UiController < ActionController::Base; end
+
+    setup { boot_dummy }
+
+    test "routes on app without routes" do
+      assert_recognizes(
+        { controller: "quilt/performance_report", action: "create" },
+        { path: "/performance_report", method: "post" }
+      )
+      assert_recognizes(
+        { controller: "quilt/ui", action: "index", path: "anything" },
+        { path: "/anything" }
+      )
+      # Use after https://github.com/rails/rails/pull/39981 is released.
+      # assert_recognizes(
+      #   { controller: "quilt/ui", action: "index" },
+      #   { path: "/" }
+      # )
+    end
+
+    test "routes on app with engine mount" do
+      Rails.application.routes.draw do
+        mount(Quilt::Engine, at: '/quilt')
+      end
+
+      assert_recognizes(
+        { controller: "quilt/performance_report", action: "create" },
+        { path: "/quilt/performance_report", method: "post" }
+      )
+      assert_recognizes(
+        { controller: "quilt/ui", action: "index", path: "anything" },
+        { path: "/quilt/anything" }
+      )
+      # Use after https://github.com/rails/rails/pull/39981 is released.
+      # assert_recognizes(
+      #   { controller: "quilt/ui", action: "index" },
+      #   { path: "/path" }
+      # )
+    end
```

Comment:
can we add another test case for a engine mount but the path for the engine is not `/` ?